src/Gui/PreferencePages/DlgSettingsEditor.cpp

```diff
@@ -327,6 +327,23 @@ void DlgSettingsEditor::loadSettings()
     ui->displayItems->setCurrentItem(ui->displayItems->topLevelItem(0));
 }
 
+void DlgSettingsEditor::resetSettingsToDefaults()
+{
+    ParameterGrp::handle hGrp;
+    hGrp = WindowParameter::getDefaultParameter()->GetGroup("Editor");
+    //reset the parameters in the "Editor" group
+    for (QVector<QPair<QString, unsigned int> >::Iterator it = d->colormap.begin(); it != d->colormap.end(); ++it) {
```

Comment:
@chennes suggestion: change the for loop with the "new style" for loop, both here and in other parts of this file. (In a new PR)